test/Microsoft.AspNetCore.Sockets.Client.Tests/ConnectionTests.cs

```diff
@@ -120,48 +140,123 @@ public class ConnectionTests
+        public async Task CannotSendAfterReceiveThrewException()
         {
+            var allowPollTcs = new TaskCompletionSource<object>();
             var mockHttpHandler = new Mock<HttpMessageHandler>();
             mockHttpHandler.Protected()
                 .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
                 .Returns<HttpRequestMessage, CancellationToken>(async (request, cancellationToken) =>
                 {
                     await Task.Yield();
+                    if (request.RequestUri.AbsolutePath.EndsWith("/poll"))
+                    {
+                        await allowPollTcs.Task;
+                        return new HttpResponseMessage(HttpStatusCode.InternalServerError) { Content = new StringContent(string.Empty) };
+                    }
                     return new HttpResponseMessage(HttpStatusCode.OK) { Content = new StringContent(string.Empty) };
                 });
 
             using (var httpClient = new HttpClient(mockHttpHandler.Object))
             using (var longPollingTransport = new LongPollingTransport(httpClient, new LoggerFactory()))
             using (var connection = await Connection.ConnectAsync(new Uri("http://fakeuri.org/"), longPollingTransport, httpClient))
             {
-                Assert.False(connection.Input.Completion.IsCompleted);
-                connection.Output.TryComplete();
+                var receiveTask = connection.ReceiveAsync(new ReceiveData());
+                allowPollTcs.TrySetResult(null);
+                await Assert.ThrowsAsync<HttpRequestException>(async () => await receiveTask);
+
+                var exception = await Assert.ThrowsAsync<InvalidOperationException>(
+                    async () => await connection.SendAsync(new byte[] { 1, 1, 3, 5, 8 }, Format.Binary));
 
-                var whenAnyTask = Task.WhenAny(Task.Delay(1000), connection.Input.Completion);
+                Assert.Equal("Cannot send messages when the connection is stopped.", exception.Message);
+            }
+        }
 
-                // The channel needs to be drained for the Completion task to be completed
-                Message message;
-                while (!whenAnyTask.IsCompleted)
+        [Fact]
+        public async Task CannotReceiveAfterReceiveThrewException()
+        {
+            var allowPollTcs = new TaskCompletionSource<object>();
+            var mockHttpHandler = new Mock<HttpMessageHandler>();
+            mockHttpHandler.Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                .Returns<HttpRequestMessage, CancellationToken>(async (request, cancellationToken) =>
                 {
-                    connection.Input.TryRead(out message);
-                    message.Dispose();
-                }
+                    await Task.Yield();
```

Comment:
What is the alternative here? This pattern has been used in the tests for a while and did not cause issues.